### PR TITLE
Slideout menu code review request

### PR DESCRIPTION
**Pull request by monalighosh**
Hi Voji,

I just finished my component,  the slideout menu. I am yet to work on a scrollbar though and some finishing touches. I would appreciate your inputs. Thank you in advance. GN.

### CODE REVIEW

**Comment by Gruximillian** [comment]
@monalighosh 
Wow, this looks nice. 👍 

But there are couple of issues present.

The biggest issue is that fo some reason the the page is limited to the veiwport height, and there's no possiblility to scroll the page. This one has to be solved for the PR to be accepted. I've looked into it just slightly now, and haven't found the cause, and probably won't be able to look more into it today. Please check it out and see if you can fix that.

The other issue is that once the menu is open it can't be closed and it is not possible to open a different sub menu.

Once those are fixed I can see that this will be an awesome component.

**Comment by Gruximillian** [comment]
@monalighosh 
I also suspect that you don't have the latest changes from dev branch. The console logs from `examples.js` are present, but in the latest that file is disabled. So, before you continue your work, make sure to update from the `dev` branch.

**Comment by OscarRobertRodriguez** [comment]
i'll take a look

**Comment by monalighosh** [comment]
Ok cool. Thanks for your inputs. Let you know once I finish the mentioned changes.

**Comment by Gruximillian** [comment]
@monalighosh 

Good, scrolling issue is resolved, but unfortunately, I found many other issues. I will go through them in detail so that you can have better insight on them and how to solve them.

### 1. Menu z position
I still cannot close the menu on some parts of the page since it is underneath the content of the rest of the page, I can open it, but then the button goes under the content to the left of it, and is not accessible anymore. Here's an image showing how the parts of the menu are under the other content:

blocked_by_content

This kind of issues are easily solved by assigning the high `z-index` to the menus that need to appear on top of the other content.

### 2. Menu container dimensions and position:
On the other hand, the menu covers the header and makes the setting button inaccessible to click:

block_content

The container has some odd positioning there. First, it should not be on the visible part of he page, but hidden as its content, and from it, the buttons can stick out to activate the content of the menu to slide left. Also, you can and should, remove `transform: translateY(-45%);` from the container and set its `top` property in a way that will enable the whole vertical content of the menu to be visible, currently it is not, its height is too large. Then set `top` property to the button labels: `top: 0`, `top: 46px` and `top: 92px`, also the slideout content should have set `top: 7px`.

This way, it won't block the header, but I'd prefer if the whole container goes off the screen, and only buttons and associated content gets animated into the page. But you can do it how every you prefer or is more confortable to you.

### 3. Menu button border right issue
There is a slight border overlap between button and the content:

border_issue

See the top border connection between the button and the content, the white right border on the button shows up, I tired and I think it is safe to add `border-right: none` on the button.

### 4. Footer issues
I assume that you this happened durig the merge and you didn't notice it. I think that the whole footer is duplicated:

footer_mess

The uppert part should not be there, fromt the title, to the links. Here is how the cleaned up code looks in the inspector:

correct_footer_structure

So, these are obvious issues that block functionality and make the appearance not good.

I have some more comments about the code itself, not much, but please don't mind me for being picky about it, I just want to us all to follow some code standards and make our code the best looking as possible. I will add those inline.

File: src/app/index.js [hunk 1]
```diff
@@ -6,9 +6,16 @@ import '../assets/images/code.jpg';
 import '../assets/images/logoBovAcademy.png'; 
 import '../assets/images/dev_news@3x.png';
 
+
+import mediaOjectTwitter from './scripts/mediaObjectTwitter'
+import slideoutMenu from './scripts/slideoutMenu'
+
+import './scripts/examples/examples';
```
Comment (Gruximillian) [inline src/app/index.js]: THis line was commented out, you can see it below. Just remove this one.

File: src/app/scripts/slideoutMenu.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+"use strict";
```
Comment (Gruximillian) [inline src/app/scripts/slideoutMenu.js]: This is nice! Everyone should add this in their code, me too. :)

File: src/index.html [hunk 3]
```diff
@@ -220,6 +220,173 @@ <h1 class="header-wrapper-item">DeveloperNews.io</h1>
       </div>
     </header>
 
+    <!-- slideout menu start -->
+    <section class="slideout-menu"> 
+      <!-- slideout menu - most popular today -->
+      <input type="checkbox" id="slideout-menu-icon-whatshot">
+      <label for="slideout-menu-icon-whatshot" class="slideout-menu__ic-whatshot">
+        <i class="material-icons slideout-menu--icons" title="Most Popular Today">whatshot</i>
```
Comment (Gruximillian) [inline src/index.html]: I was being a pain for everyone on this. :)
Please don't use the `i` element for the icons, you know that its purpose is not to display images. The regular `span` can be used without any issues. Using `i` goes against the html standards.

File: src/styles/layout/_slideoutMenu.scss [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+     width: 100%;
+     max-width: 270px;
+     height: 520px;
+     overflow: hidden;
+     transition: all 0.5s ease;
+    }
+
+    &__content-heading {
+     font-weight: 500;
+     font-size: $baseFontSize*1.5;
```
Comment (Gruximillian) [inline src/styles/layout/_slideoutMenu.scss]: Please add spaces around the multiplication operator: `$baseFontSize * 1.5`

File: src/styles/layout/_slideoutMenu.scss [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+    &__content-list {
+     list-style-type: none;
+     padding: 0;
+    }
+
+    &__content-list-item {
+     font-weight: 500;
+     padding: 0.8rem 0;  	
+     font-size: $baseFontSize*1.07;
```
Comment (Gruximillian) [inline src/styles/layout/_slideoutMenu.scss]: Please add spaces around the multiplication operator: `$baseFontSize * 1.5`

File: src/styles/layout/_slideoutMenu.scss [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+
+    &__content-list-item-title {
+      padding: 0.3rem 0.5rem;  
+      position: relative;
+      border: 1px solid $red;
+      margin: 1.3rem 0;	
+    }
+
+    &__gradient-bg {
+      background: linear-gradient(to bottom, rgba(255,0,0,0), rgba(255,255,255,1));
```
Comment (Gruximillian) [inline src/styles/layout/_slideoutMenu.scss]: Please add spaces after the comma inside the rgba values.

File: src/styles/layout/_slideoutMenu.scss [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    &--icons-list-item {
+     color: darken($lightGray, 15%);  
+     position: absolute;
+     right: 0;
+     top: 0; 
+     margin-top: 13px; 
+     transition: all 0.5s ease;	
+
+       &:hover {
+       	color: $gray;
```
Comment (Gruximillian) [inline src/styles/layout/_slideoutMenu.scss]: Please make the indentation in the whole file 2 spaces. You have one on many lines and I thin that you have a tab on this one.

File: src/styles/main.scss [hunk 8]
```diff
@@ -7,7 +7,11 @@
 @import 'layout/header';
 @import 'layout/submitArticle';
 @import 'layout/topNews';
+
```
Comment (Gruximillian) [inline src/styles/main.scss]: No need for empty lines, group those layout imports.

**Comment by Gruximillian** [comment]
@monalighosh 
Ok, I've added all comments, and have one suggestion.
Currently, when the menu is open you can't click on the menu button for a different section, you must close the entire menu and reopen by clicking on the different section. It would be nice if user could open it without closeing the menu previously.

This is an awesome component, all the issues are very small ones, but they add up and make it unsusable, when you deal with them it wil be awesome. Keep it up!

**Comment by monalighosh** [comment]
Hey, I have resolved both the issues. I have incorporated a close button into the content window. Also pulled all the updates from dev branch. Hope its fine now. Kindly have a look and let me know if there is anything. Thank you in advance.

**Comment by monalighosh** [comment]
Sorry, I didn't notice your recent comments. I am working on those issues.

**Comment by monalighosh** [comment]
Hey voji, don't feel that way. I like you being picky. Constructive criticism is always welcomed. I have fixed all the issues mentioned above. Kindly chk and let me know if there is anything. Thank you.

**Comment by Gruximillian** [comment]
@monalighosh 

Thank you for these changes. I LOVE how the menu switches its submenus, so lively, I just want to click on the buttons to watch it switch. :)

There's still the positioning issue, I am not sure, but do you use a screen with large height? The menu is still at a low position on the screen and not completely visible. You have this rule: `.slideout-menu {top: 600px;}` which is too much, if you go with something around 200px it will be perfect, just under the header and it will fit on the view.

Another issue appeared now that you added the close button. If you are somewhere down the page, and press the close button, the view jumps to the top of the page. You are using the `a href="#"`, the # is causing this. I think you could use a `div` or a `span` for this or even ` button`, you don't need the `a` element.

Anyway, I'll merge this now since the page works, these issues are not too big now. You can make these changes afterwards.

**Pull request merged**